### PR TITLE
Fixes for file-watching, user assets, and system dependencies

### PR DESCRIPTION
**Pull request by aminya**
https://github.com/zed-industries/zed/pull/7343

### CODE REVIEW

**Review by williamdes — APPROVED** [review]

**Review by kvark — APPROVED** [review]

File: script/linux [hunk 1]
```diff
@@ -21,6 +28,7 @@ if [[ -n $apt ]]; then
   deps=(
     libasound2-dev
     libfontconfig-dev
+    vulkan-validationlayers*
```
Comment (kvark) [inline script/linux:31]: Are we sure we need this for everyone? It's only required for developers, see https://github.com/zed-industries/zed/pull/7343/files#diff-e6075538b36d33b199cdfd03cfae8988bdcb787d012b369eca2a3607f95b239eR199

**Review by kvark — COMMENTED** [review]
Happy to merge the first 2 commits but have a question about the validation layers

File: script/linux (hunk 1 above)
Comment (aminya) [reply script/linux:31]: We can make it configurable before the final merge to only be used during the debug builds. 
For now, no one can use these builds yet, so it makes things safer to catch the bugs.

File: script/linux [hunk 2]
```diff
@@ -3,10 +3,17 @@
...
 [[ $(uname) == "Linux" ]] || exit 0
 
-# Copy settings and keymap to the user's home directory if they don't exist
+# Copy assets to the user's home directory if they don't exist
 mkdir -p "$HOME/.config/zed"
+
+mkdir -p "$HOME/.config/zed/plugins"
+
+mkdir -p "$HOME/.config/zed/themes"
+cp -ruL ./assets/themes/*/*.json "$HOME/.config/zed/"
```
Comment (nhemsley) [inline script/linux]: @aminya This copies into ~/.config/zed, not into ~/.config/zed/themes.

File: script/linux (hunk 1 above)
Comment (kvark) [reply script/linux:31]: sounds good!

File: script/linux (hunk 2 above)
Comment (aminya) [reply script/linux]: Good catch. Fixed in [`73fe59b` (#2)](https://github.com/kvark/zed/pull/2/commits/73fe59b692295e8ee3b517e05c79b28b8261b566)

**Comment by aminya** [comment]
@kvark Rebased and fixed the conflicts.

**Review by kvark — APPROVED** [review]

**Pull request merged**